packages/contents/src/drivefs.ts

```diff
@@ -550,24 +555,28 @@ export abstract class ContentsAPI {
  * An Emscripten-compatible synchronous Contents API using the service worker.
  */
 export class ServiceWorkerContentsAPI extends ContentsAPI {
-  constructor(
-    baseUrl: string,
-    driveName: string,
-    mountpoint: string,
-    FS: FS,
-    ERRNO_CODES: ERRNO_CODES,
-  ) {
-    super(driveName, mountpoint, FS, ERRNO_CODES);
+  /**
+   * Construct a new ServiceWorkerContentsAPI.
+   */
+  constructor(options: ServiceWorkerContentsAPI.IOptions) {
+    super(options);
 
-    this._baseUrl = baseUrl;
+    this._baseUrl = options.baseUrl;
+    this._browsingContextId = options.browsingContextId || '';
   }
 
   request<T extends TDriveMethod>(data: TDriveRequest<T>): TDriveResponse<T> {
     const xhr = new XMLHttpRequest();
     xhr.open('POST', encodeURI(this.endpoint), false);
 
+    // Add the origin browsing context ID to the request
+    const requestWithBrowsingContextId = {
+      ...data,
+      browsingContextId: this._browsingContextId,
+    };
+
```

Comment:
Do you think embedding the context ID via the request header is slightly cleaner? It helps avoid polluting the payload data.

Reply:
Indeed, so maybe we could keep data as an object key and not spread it.

Reply:
Wouldn't keeping `data` collide with the existing `data` field used for some of the methods?

https://github.com/jupyterlite/jupyterlite/blob/b01f12f02ccc18aa0ab988e8162d9b4fc57cf93d/packages/contents/src/drivefs.ts#L87

For example with `mknod`:

![image](https://github.com/user-attachments/assets/74ec4738-6f54-48d3-86b0-5ca89cba2481)

Reply:
i meant 

```ts
    const requestWithBrowsingContextId = {
      data,
      browsingContextId: this._browsingContextId,
    };
```

Reply:
But it's just a minor detail. The remaining logic looks good

Reply:
ok I just pushed https://github.com/jupyterlite/jupyterlite/pull/1621/commits/0d009b4a9d04698f403f4ea1a747ddf71f91e393 to do something along those lines.